### DISCUSSION TITLE
Grafana Tempo on windows

### DISCUSSION THREAD

**Discussion by srinivasmummareddy** [grafana/tempo#3390] [General]
Is there any way to install tempo on windows?(I don't want to use docker)

**Comment by joe-elliott** [comment]
We provide a compiled windows binary on every release. You should be able to run this on windows, but I don't think many run it this way so it may have some unknown issues.

https://github.com/grafana/tempo/releases

**Comment by srinivasmummareddy** [comment]
@joe-elliott Is there any documentation for how to configure  this like storage, receiving traces and visualization?

**Comment by joe-elliott** [comment]
I would refer to a docker compose such as this one:

https://github.com/grafana/tempo/blob/main/example/docker-compose/s3/docker-compose.yaml

to see how the config can be used to setup Tempo and connect it to a grafana. that particular one uses s3. if you'd rather just store data locally for a test try this one:

https://github.com/grafana/tempo/blob/main/example/docker-compose/local/docker-compose.yaml

**Comment by srinivasmummareddy** [comment]
@joe-elliott  Is there any specific documentation for windows? Like how to pass config file path to tempo?

level=info ts=2024-02-15T04:35:03.2866634Z caller=main.go:222 msg="initialising OpenTracing tracer"
level=info ts=2024-02-15T04:35:03.4298755Z caller=main.go:118 msg="Starting Tempo" version="(version=2.3.1, branch=HEAD, revision=9398fd620)"
level=error ts=2024-02-15T04:35:03.4298755Z caller=main.go:121 msg="error running Tempo" err="failed to init module services: error initialising module: store: failed to create store: unknown backend "

I'm getting this  above error

**Comment by joe-elliott** [comment]
Did you try to load one of the suggested config files? A minimal setup is something like:
```
go run ./cmd/tempo --storage.trace.backend=local --storage.trace.local.path=/tmp/tempo/traces --storage.trace.wal.path=/tmp/tempo/wal
```

you will likely also need to set the http port b/c it will attempt to open 80 by default which windows probably won't like.

there is no specific windows docs.

**Comment by m0a0k0s** [comment]
Could it load a configuration file, instead of all settings one-by-one?

**Reply by m0a0k0s** [reply]
found the solution
tempo.exe -config.file=tempo.yml

**Reply by admin-aquaware** [reply]
Hey I got a question, Did your tempo Installation on Windows work ? Or did you have issues after that?

**Reply by m0a0k0s** [reply]
It worked, but it was poc with small amount of data. Already migrated to linux.
